budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -172,42 +342,150 @@ def test_filter_by_funding_source(self):
+
+        data = {
+            "selected_facets": "primary_funding_source_exact:{0}".format(new_source)
+        }
+        response = self.client.get(self.facet_url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        province_facets = response.data["fields"]["province"]
+        provinces_after_filtering = 0
+        for value in province_facets:
+            if province == value["text"]:
+                provinces_after_filtering = value["count"]
+
+        self.assertEqual(provinces_after_filtering, 3)
+
     def test_search_by_project_name(self):
         name = "Project 1"
-        data = {"search": name}
+        data = {"q": name}
         response = self.client.get(self.url, data)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertContains(response, name)
+
+        results = response.data["results"]
+        self.assertEqual(len(results), 1)
+        self.assertEqual(results[0]["name"], name)
+
+    def test_facet_search_by_project_name(self):
+        name = "Project 1"
+        data = {"q": name}
+        response = self.client.get(self.facet_url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        results = response.data["objects"]["results"]
+        self.assertEqual(len(results), 1)
+        self.assertEqual(results[0]["name"], name)
 
     def test_search_by_municipality(self):
+        name = "Project 1"
         municipality = "Local 1"
-        data = {"search": municipality}
+        data = {"q": municipality}
         response = self.client.get(self.url, data)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertContains(response, municipality)
+
+        results = response.data["results"]
+        self.assertEqual(len(results), 1)
+        self.assertEqual(results[0]["name"], name)
+
+    def test_facet_search_by_municipality(self):
+        municipality = "Local 1"
+        data = {"q": municipality}
+        response = self.client.get(self.facet_url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        results = response.data["objects"]["results"]
+        self.assertEqual(len(results), 1)
+
+        province = "Eastern Cape"
+        data = {"selected_facets": "province_exact:{0}".format(province)}
+        response = self.client.get(self.facet_url, data)
+        province_results = response.data["objects"]["results"]
+        self.assertNotEqual(results, province_results)
 
     def test_search_by_province(self):
         province = "Eastern Cape"
-        data = {"search": province}
+        data = {"q": province}
         response = self.client.get(self.url, data)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertContains(response, province)
+
+        results = response.data["results"]
+        self.assertEqual(len(results), 15)
+
+    def test_facet_search_by_province(self):
```

Comment:
`search_by_` should use  `q` to use _full text search_ rather than _filter_

Reply:
I was using ```q``` but I think I forgot it in one search method. Anyway, I corrected it

Reply:
Thanks - yeah I also think it was just this one